Support API level 29 (Android 10)

# Feature Request

## Motivation Behind Feature
Because android 10 is announced and released, platforms should be updated to version 29 when I do `cordova build android`

https://developer.android.com/studio/releases/platforms

Currently, the API version supported with android-cordova 8.1.0 is 19-28 for android 4.4-9.0.0 

https://cordova.apache.org/docs/en/latest/guide/platforms/android/

## Alternatives or Workarounds
Using android studio directly to release and update platforms to build my aab, but I'm using docker to build the aab for automated operations.

@breautek 
I tried that, still got:

```console
Command finished with error code 1: /app/platforms/android/gradlew :app:bundleRelease,-b,/app/platforms/android/build.gradle
/app/platforms/android/gradlew: Command failed with exit code 1 Error output:
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> Failed to install the following SDK components:
      platforms;android-28 Android SDK Platform 28
  Install the missing components using the SDK manager in Android Studio.


```

> @breautek
> I tried that, still getting:
> 
> ```
> Command finished with error code 1: /app/platforms/android/gradlew :app:bundleRelease,-b,/app/platforms/android/build.gradle
> /app/platforms/android/gradlew: Command failed with exit code 1 Error output:
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> A problem occurred configuring project ':app'.
> > Failed to install the following SDK components:
>       platforms;android-28 Android SDK Platform 28
>   Install the missing components using the SDK manager in Android Studio.
> ```

I am unable to reproduce this behaviour with a very simple node app. Cordova finds my build tools sdk `29.0.0` just fine & it obeys the `android-targetSdkVersion` just fine for me. I added additonal print lines in my project gradle file to see which build tools sdk and target SDK it is using which I've pasted below.

```
> Configure project :app
BUILD TOOLS VERSION
29.0.0
TARGET SDK VERSION
DefaultApiVersion{mApiLevel=29, mCodename='null'}
...
```

I would suggest making sure everything is up to date, including the cordova CLI.

`npm install cordova@9 -g`

Also do ensure that your `ANDROID_HOME` system environment variable is set properly.

Sadly, still not working:

```console
Checking the license for package Android SDK Platform 28 in /opt/android-sdk-linux/licenses
License for package Android SDK Platform 28 accepted.
Preparing "Install Android SDK Platform 28 (revision: 6)".
Warning: Failed to read or create install properties file.

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> Failed to install the following SDK components:
      platforms;android-28 Android SDK Platform 28
  Install the missing components using the SDK manager in Android Studio.


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3m 22s
Command finished with error code 1: /app/platforms/android/gradlew :app:bundleRelease,-b,/app/platforms/android/build.gradle
/app/platforms/android/gradlew: Command failed with exit code 1 Error output:
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> Failed to install the following SDK components:
      platforms;android-28 Android SDK Platform 28
  Install the missing components using the SDK manager in Android Studio.


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

```

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,8 +7,8 @@ dist: trusty
 
 env:
   global:
-    - ANDROID_API_LEVEL=28
-    - ANDROID_BUILD_TOOLS_VERSION=28.0.3
+    - ANDROID_API_LEVEL=29
+    - ANDROID_BUILD_TOOLS_VERSION=29.0.2
     - TERM=dumb # Keep gradle from crapping all over the log
   matrix:
     - nodejs_version=6

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -21,7 +21,7 @@ install:
     - set PATH=%PATH%;"%ANDROID_HOME%\tools\bin"
 
     - yes 2> nul | sdkmanager --licenses > nul
-    - sdkmanager "build-tools;28.0.3"
+    - sdkmanager "build-tools;29.0.2"
 
     - choco install gradle --version 3.4.1
 

--- a/bin/templates/project/build.gradle
+++ b/bin/templates/project/build.gradle
@@ -40,10 +40,10 @@ allprojects {
 
     //This replaces project.properties w.r.t. build settings
     project.ext {
-      defaultBuildToolsVersion="28.0.3" //String
+      defaultBuildToolsVersion="29.0.2" //String
       defaultMinSdkVersion=19 //Integer - Minimum requirement is Android 4.4
-      defaultTargetSdkVersion=28 //Integer - We ALWAYS target the latest by default
-      defaultCompileSdkVersion=28 //Integer - We ALWAYS compile with the latest by default
+      defaultTargetSdkVersion=29 //Integer - We ALWAYS target the latest by default
+      defaultCompileSdkVersion=29 //Integer - We ALWAYS compile with the latest by default
     }
 }
 

--- a/framework/project.properties
+++ b/framework/project.properties
@@ -5,7 +5,7 @@
 split.density=false
 
 # Project target.
-target=android-28
+target=android-29
 apk-configurations=
 renderscript.opt.level=O0
 android.library=true

--- a/spec/fixtures/android_studio_project/app/build.gradle
+++ b/spec/fixtures/android_studio_project/app/build.gradle
@@ -1,13 +1,13 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 28
-    buildToolsVersion "28.0.3"
+    compileSdkVersion 29
+    buildToolsVersion "29.0.2"
 
     defaultConfig {
         applicationId "com.example.anis.myapplication"
         minSdkVersion 21
-        targetSdkVersion 28
+        targetSdkVersion 29
         versionCode 1
         versionName "1.0"
     }

--- a/test/app/build.gradle
+++ b/test/app/build.gradle
@@ -19,13 +19,13 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 28
-    buildToolsVersion "28.0.3"
+    compileSdkVersion 29
+    buildToolsVersion "29.0.2"
 
     defaultConfig {
         applicationId "org.apache.cordova.unittests"
         minSdkVersion 19
-        targetSdkVersion 28
+        targetSdkVersion 29
         versionCode 1
         versionName "1.0"
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"

